test/built-ins/Temporal/Calendar/prototype/yearOfWeek/argument-string-date-with-utc-offset.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.yearofweek
+description: UTC offset not valid with format that does not include a time
+features: [Temporal]
+---*/
+
+const instance = new Temporal.Calendar("iso8601");
+
+const validStrings = [
+  "2000-05-02T00+00:00",
+  "2000-05-02T00+00:00[UTC]",
+  "2000-05-02T00+00:00[!UTC]",
+  "2000-05-02T00-02:30[America/St_Johns]",
+];
+
+for (const arg of validStrings) {
+  const result = instance.yearOfWeek(arg);
+
+  assert.sameValue(
+    result,
+    2000,
+    `"${arg}" is a valid UTC offset with time for PlainDate`
+  );
+}
+
+const invalidStrings = [
+  "2022-09-15Z",
+  "2022-09-15Z[UTC]",
+  "2022-09-15Z[Europe/Vienna]",
+  "2022-09-15+00:00",
+  "2022-09-15+00:00[UTC]",
+  "2022-09-15-02:30",
+  "2022-09-15-02:30[America/St_Johns]",
+];
+
+for (const arg of invalidStrings) {
+  assert.throws(
+    RangeError,
+    () => instance.yearOfWeek(arg),
+    `"${arg}" UTC offset without time is not valid for PlainDate`
```

Comment:
Is this just testing that Temporal.Calendar.prototype.yearOfWeek internally converts its argument to a PlainDate? If so, then I'd rather see that made clear and a standard suite of valid and invalid values presented to it, because most if not all other Calendar prototype methods are _also_ expected to perform such conversion, as are various other operations:
* Temporal.PlainDate.{from,compare}
* Temporal.PlainDate.prototype.{equals,since,until}
* Temporal.PlainTime.prototype.to{Plain,Zoned}DateTime
* the various `withPlainDate` methods
* {Balance,Unbalance}DurationRelative
* RoundDuration

Reply:
By standard suite of valid and invalid values do you mean something like the arrays in https://github.com/tc39/test262/blob/e7895c53d295760d7c2b89743f8311fa60480392/harness/temporalHelpers.js#L1800? I agree that would be better, but I'd prefer to keep that out of scope for this PR. I'd just like this to be a quick fix for the discrepancy between #3696 and #3699.

Reply:
I do mean that, and agree on the scope of this PR. :+1: